DistroLauncher/OOBE.cpp

```diff
@@ -161,23 +164,68 @@ namespace DistributionInfo {
 				return S_OK;
 			}
 
+			// Before relaunching, give WSL some time to make sure 
+			// distro is stopped.
+			bool stopSuccess = EnsureStopped(30u);
+			if (!stopSuccess) {
+				// We could try again, but who knows why
+				// we failed to stop the distro in the first time.
+				return ERROR_FAIL_SHUTDOWN;
+			}
+
 			// We could, but may not want to just `wsl -d Distro`.
 			// We can explore running our launcher in the future.
 			TCHAR launcherName[MAX_PATH];
 			DWORD fnLength = GetModuleFileName(NULL, launcherName, MAX_PATH);
 			if (fnLength == 0) {
 				return HRESULT_FROM_WIN32(GetLastError());
 			}
-			// Just to ensure we respect the returned buffer size:
-			const std::wstring_view launcher{ launcherName, fnLength };
-			cmdResult = _wsystem(launcher.data());
+
+			cmdResult = _wsystem(launcherName);
 			if (cmdResult != 0) {
 				return ERROR_FAIL_RESTART;
 			}
 
 			return S_OK;
 		} // HRESULT OOBEStatusHandling(std::wstring_view status).
 
+		// Polls WSL to ensure the distro is actually stopped.
+		bool EnsureStopped(unsigned int timeoutSeconds) {
```

Comment:
it’s technically here more a number of retry from the code. It happens to be matching the second due to the sleep() call being almost one sec, but it may be modified later on, which will make this variable lying in future :p

Reply:
Would you recommend renaming it to `maxNoOfAttempts` ?

Reply:
or `maxRetries`? Both works for me, pick your preference :)